app/models/course/discussion/post.rb

```diff
@@ -172,4 +235,29 @@ def mark_self_as_read
   def sanitize_text
     self.text = ApplicationController.helpers.sanitize_ckeditor_rich_text(text)
   end
+
+  def ensure_rag_auto_answering!
+    ActiveRecord::Base.transaction(requires_new: true) do
+      rag_auto_answering || create_rag_auto_answering!
+    end
+  rescue ActiveRecord::RecordInvalid, ActiveRecord::RecordNotUnique => e
+    raise e if e.is_a?(ActiveRecord::RecordInvalid) && e.record.errors[:post_id].empty?
+
+    association(:rag_auto_answering).reload
+    rag_auto_answering
+  end
+
+  def create_topic_subscription(topic, current_user)
+    if topic.forum.forum_topics_auto_subscribe
+      topic.ensure_subscribed_by(current_user)
+    else
+      true
+    end
+  end
+
+  def send_created_notification(current_author, current_course_author)
+    return unless current_author
+
+    Course::Forum::PostNotifier.post_replied(current_author, current_course_author, self)
```

Comment:
How come this logic for `send_created_notification` and `create_topic_subscription` moved from controller/course/forum/post_controller into models/course/discussion/post?

Reply:
I need to call `send_created_notification `and `create_topic_subscription` during text_chunk job. Thus, I shifted the logic to model as I would be unable to call them from the job if they are in the controller.

Reply:
Shifted the logic into auto_answering_concern.rb file